### PR TITLE
Adding docker compose

### PR DESCRIPTION
**Pull request by ratneshraval**
#### Which User story does this relate to?
- https://www.pivotaltracker.com/story/show/151624427

#### Brief feature description
- Setup to run rspec inside container.

#### Does this PR block/effect other stories?
- Affects Jenkins PR testing

### CODE REVIEW

File: lib/tasks/spec.rake [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+end
+
+namespace :spec do # rubocop:disable BlockLength
+  def file_list
+    # first ARGV is task name
+    args = ARGV.drop(1)
+    args.any? ? args.join(' ') : 'spec'
+  end
+
+  def webpack?
```
Comment (lavanyareddyreddy) [inline lib/tasks/spec.rake]: I think this can be renamed because its not returning a boolean

Comment (ratneshraval) [reply lib/tasks/spec.rake]: Updated.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# 6/23/2017 - version 3.x does not support extends yet.
+# - https://docs.docker.com/compose/extends/#extending-services
+services:
+  cals:
+    build: .
+    command: bash -c "bundle check || bundle install && yarn && bundle exec puma"
+    ports:
+      - "3000:3000"
+      - "4857:4857"
+    environment:
```
Comment (anuroop15) [inline docker-compose.yml]: @ratneshraval : if docker file shows all of our env files, doesn't our env files get exposed ?

Comment (ratneshraval) [reply docker-compose.yml]: It is ok for test setup.

**Pull request merged**